packages/utils/regl-renderer/src/geometry-utils-V2/geom2ToGeometries.js

```diff
@@ -17,16 +17,36 @@ const geom2ToGeometries = (options, solid) => {
 
   if ('color' in solid) color = solid.color
   const isTransparent = (color[3] < 1.0)
-
+  const colors = []
   const numgeometries = Math.floor(sides.length / (maxIndex)) + 1
+  // make sure color has alpha channel
+  const addColor = (c) => {
+    // add alpha channel (copy, do not mutate)
+    if (c.length === 3) c = [...c, 1]
```

Comment:
colorize() from modeling guarantees that colors are RGBA.
and this is the renderer sitting on top of WEBGL, which requires RGBA colors.
so, i think we should be strict and throw an error if colors are not RGBA.

@platypii what do you think?

Reply:
you may have noticed that I support this reporting errs early and forcing users to provide clean input to avoid excessive pre-emptive fixing of provided input. 

Benefit of having 3 channel colors is reduced memory footprint (but only in case of color-per-vertex count is large enough to matter). 
This benefit may not be significant enough to allow 3 channel colors in jscad modeling. Even if jscad modeling will use 4 channel colors, the code converting to webgl buffers could decide to use shader with 3 colors per channel for opaque mesh.

WEBGL internally uses RGBA, but there are some performance benefits of knowing if colors have transparency on a geometry so blending can be skipped for opaque geoms. It is far more important for code that converts meshes to data for webgl to provide information for each mesh if it has transparency. This allows for turnigin off blending.